src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,76 @@
+import cn from 'classnames';
+import { FilterState } from '../../types/FilterStates';
+
+interface FooterProps {
+  itemsLeft: number;
+  filterState: FilterState;
+  onFilter: (filterState: FilterState) => void;
+  disableClearButton: boolean;
+  onClearCompleted: () => void;
+}
+
+export const Footer: React.FC<FooterProps> = ({
+  itemsLeft,
+  filterState,
+  onFilter,
+  disableClearButton,
+  onClearCompleted,
+}) => {
+  return (
+    <>
+      {/* Hide the footer if there are no todos */}
+      <footer className="todoapp__footer" data-cy="Footer">
+        <span className="todo-count" data-cy="TodosCounter">
+          {itemsLeft} items left
+        </span>
+
+        {/* Active link should have the 'selected' class */}
+        <nav className="filter" data-cy="Filter">
+          <a
+            href="#/"
+            className={cn('filter__link', {
+              selected: filterState === 'All',
+            })}
+            data-cy="FilterLinkAll"
+            onClick={() => onFilter('All')}
+          >
+            All
+          </a>
+
+          <a
+            href="#/active"
+            className={cn('filter__link', {
+              selected: filterState === 'Active',
+            })}
+            data-cy="FilterLinkActive"
+            onClick={() => onFilter('Active')}
+          >
+            Active
+          </a>
+
+          <a
+            href="#/completed"
+            className={cn('filter__link', {
```

Comment:
Use `Object.values(your created enum)` and render these options with `map()` method